### ISSUE TITLE
Having multiple ng2 modules in an Upgrade scenario: it is possible?

### ISSUE THREAD

**Issue by lmartorella** [angular/angular#14671]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[X] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

* **Angular version:** 2.4.8
* **Browser:** all
* **Language:** all

**Current behavior**
It seems that multiple Angular2 modules cannot downgrade directives to ng1 at the same time.
The documentation seems to be not clear on that. How it is possible to downgrade directives coming from different modules?

**What is the motivation / use case for changing the behavior?**
Our legacy application is an Angular 1.x shell containing multiple applicative angular modules.
Now we are planning a progressive migration of applicative modules to Angular 2, leaving the outer shell the last one to migrate.
However we are striving to load multiple Angular 2 module in an Angular 1.x app following the Upgrading guidelines. 

**Minimal reproduction of the problem with instructions**
Here a minimum plunker with our scenario: http://plnkr.co/edit/nBRFJBDDZRiR6jmiO3LA
Basically there are one 'Core' module and two 'App' modules, each one defining one component. All the components has to be rendered in the Angular1 shell (registered through downgradeComponent).
The expected output should be to see all three components (core, app1 and app2).

We receive an error instead when the injector (coming from ng1) is trying to find the componentfactory for AppComponent1. The injector used there is the Core app one.

Thank you,
 Luciano

**Comment by gkalpak** [comment]
You are essentially bootstrapping three Angular mini-apps. In this case, each one will have its own injector and compiler, than don't know anything about the components and providers in the other apps (thus the error you get).

Although an unusual usecase, you could bootstrap one AngularJS mini-app inside each Angular mini-app, so that injectors/compilers know what they need for each app ([updated demo][1]).

[1]: http://plnkr.co/edit/J3qXQlf2HFnY4s5kwjpX?p=preview

**Comment by lmartorella** [comment]
Well, your scenario suppose that the mini-apps doesn't interact each other.
However our AngularJS structure was different: one "core" module that acts as shell and many "app" modules that are implementing business widgets.
Let's think to a composite UI like Visual Studio or Eclipse: the core module offer the docking manager, menu bar, etc... and the app modules are adding business objects to create within the core/shell.

In case of progressive migration of ng1 app modules to Angular2/5, we cannot assign to each mini-app a DOM root like in your sample, since all the apps are sharing the same root offered by the core/shell module.

Could you please to update the above plunkr in order to allow migration of app modules but bootstrap a common root?
Thanks,
 Luciano
